### PR TITLE
Hide doafters if you're in a container

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Hides do-after bars if you're in a container

## Why / Balance

Stealth gameplay is good, do-after just gives it away through meta information

## Technical details

Checks if the person performing the do-after is inside a container, hides it from view of other players. You can still see your own do-after bar.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/58439124/9a48da8d-4d6c-4e64-8a79-d53ef99bfd6d

## Breaking changes

**Changelog**

:cl:
- add: Do-after bars performed inside containers are hidden from view.

### CODE REVIEW

**Comment by Sirionaut** [comment]
What about situations where someone is trapped in a welded locker/crate and tries to break out, should that be invisible to everyone around?

**Comment by Plykiya** [comment]
> What about situations where someone is trapped in a welded locker/crate and tries to break out, should that be invisible to everyone around?

I don't see why it would be visible, but regardless, if the person wants to be found by other people they can also just talk.

File: Content.Client/DoAfter/DoAfterOverlay.cs [hunk 1]
```diff
@@ -102,7 +105,7 @@ protected override void Draw(in OverlayDrawArgs args)
             {
                 // Hide some DoAfters from other players for stealthy actions (ie: thieving gloves)
                 var alpha = 1f;
-                if (doAfter.Args.Hidden)
+                if (doAfter.Args.Hidden || _container.IsEntityInContainer(uid))
                 {
```
Comment (metalgearsloth) [inline Content.Client/DoAfter/DoAfterOverlay.cs:109]: This is going to be checked inside of the most expensive loop when you should be checking it above + also I believe you can pass metadatacomponent in and more.

**Comment by Brandon-Huu** [comment]
> I don't see why it would be visible,

Me smashing my torso against the door somehow not visible to others?

Also it should show a do-after if the action is being performed on you. ( captain should know if someone who is right next to them is taking their clothing )

**Comment by Tayrtahn** [comment]
> What about situations where someone is trapped in a welded locker/crate and tries to break out, should that be invisible to everyone around?

Seeing a progress bar for that from the outside is kinda weird, but there definitely should be some indication that someone is trying to break it, since it's not a subtle action. 

Having the locker/crate shake and make banging noises would be cooler.

**Comment by deltanedas** [comment]
steal grinder shaking but less intense

**Comment by Plykiya** [comment]
> > What about situations where someone is trapped in a welded locker/crate and tries to break out, should that be invisible to everyone around?
> 
> Seeing a progress bar for that from the outside is kinda weird, but there definitely should be some indication that someone is trying to break it, since it's not a subtle action.
> 
> Having the locker/crate shake and make banging noises would be cooler.

yeah there's not even a sound effect for it right now, it's kinda weird

**Pull request merged**